server/tests/src/main/scala/sttp/tapir/server/tests/ServerSecurityTests.scala

```diff
@@ -108,6 +110,36 @@ class ServerSecurityTests[F[_], S, OPTIONS, ROUTE](createServerTest: CreateServe
         .serverLogic(s => _ => pureResult(s.asRight[Unit]))
     ) { (backend, baseUri) =>
       basicStringRequest.get(uri"$baseUri/auth").auth.bearer("1234").send(backend).map(_.body shouldBe "1234")
+    },
+    testServerLogic(
+      in_security_option_basic_out_string
+        .serverSecurityLogic((u: Option[UsernamePassword]) => pureResult(u.map(_.username).getOrElse("noauth").asRight[Unit]))
+        .serverLogic(s => _ => pureResult(Right(s))),
+      "In security Option[UsernamePassword] should let in basic auth as well as request without auth"
+    ) { (backend, baseUri) =>
+      (for {
+        basicAuth <- basicStringRequest.get(uri"$baseUri/auth").auth.basic("a", "b").send(backend)
+        withoutAuth <- basicStringRequest.get(uri"$baseUri/auth").send(backend)
+      } yield (basicAuth, withoutAuth)).map { case (basic, without) =>
+        (basic.code.code -> without.code.code) shouldBe ((200, 200))
```

Comment:
you can just do
```
basic.code.code shouldBe 200
without.code.code shouldBe 200
```

scalatest is imperative and simply throws an exception when an assertion fails